ui/desktop/tests/end2end/authentication.spec.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable no-undef */
+const { test, expect } = require('@playwright/test');
+const { _electron: electron } = require('playwright');
+const path = require('path');
+
+let electronApp = null;
+const returnScreenshotPath = (fileName) => {
+  const screenshotPath = 'tests/end2end/screenshots/authentication/';
+  const screenshotFormat = '.png';
+  return path.join(screenshotPath, fileName).concat(screenshotFormat);
+};
+
+test.beforeEach(async () => {
+  electronApp = await electron.launch({
+    executablePath:
+      'electron-app/out/Boundary-darwin-arm64/Boundary.app/Contents/MacOS/Boundary',
+    env: {
+      NODE_ENV: 'test',
+      BYPASS_APP_UPDATER: true,
+    },
+  });
+});
+
+test.afterEach(async () => {
+  electronApp.close();
+  electronApp = null; // Not sure we should do this.
+});
+
+test.describe('Authentication end to end test suite', async () => {
+  test('Authenticates using user and password method and deauthenticates  ', async () => {
+    const boundaryWindow = await electronApp.firstWindow(); // The window that contains the app.
+    // Override local storage origin
+    await boundaryWindow.evaluate(() =>
+      window.localStorage.setItem('desktop:origin', null)
+    );
+    const originValue = 'http://localhost:9200';
+    const authLoginNameValue = 'admin';
+    const authLoginPasswordValue = 'password';
+
+    // Fill the origin input
+    await boundaryWindow.fill('.ember-text-field', originValue);
+    // Take screenshot
+    await boundaryWindow.screenshot({
+      path: returnScreenshotPath('fillOrigin'),
+      fullPage: true,
+    });
+
+    // Click the submit button
+    // Due to an error with await boundaryWindow.click('button[type="submit"]'); we are using a workaround.
+    // More info about it here: https://github.com/microsoft/playwright/issues/1808
+    await boundaryWindow.$eval('button[type="submit"]', (element) =>
+      element.click()
+    );
+    // Fill user & password
+    await boundaryWindow.fill('[name="identification"]', authLoginNameValue);
+    await boundaryWindow.fill('[name="password"]', authLoginPasswordValue);
+    // Take screenshot
+    await boundaryWindow.screenshot({
```

Comment:
In the future I do plan to minimize repeating code to take screenshots. In the future I will simplify this with a function call, but for now I do not want to complicated it.